core/controllers/admin.py

```diff
@@ -854,3 +856,26 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+        if new_username is not None:
+            if len(new_username) > MAX_USERNAME_LENGTH:
+                raise self.InvalidInputException(
+                    'Please make sure that the new username is not longer '
+                    'than %s characters.' % MAX_USERNAME_LENGTH)
+            if user_services.is_username_taken(new_username):
+                raise self.InvalidInputException('Username already taken.')
+        user_id = user_services.get_user_id_from_username(current_username)
```

Comment:
Check this before calling `is_username_taken`

Reply:
Done.